llama_stack/providers/inline/tool_runtime/rag/memory.py

```
@@ -127,7 +129,7 @@ async def query(
         scores = [s for r in results for s in r.scores]
 
         if not chunks:
-            return RAGQueryResult(content=None)
+            raise ValueError("The knowledge search tool did not find any information relevant to the query.")
```

why is not retrieving relevant info an error?

---

For two reasons:
1. Assuming we're dealing with standard vector DBs, some kind of similarity search always takes place upon retrieval. No matter what the query is, there will always be some chunk(s) with the smallest distance in the embedding space - unless the DB is empty which is covered by the previous error.
2. The way it works now, returning None as the content still results in an error message on a higher level (see above). It can be further discussed whether this behavior makes sense for a generic tool, but this discussion would be out of scope of this PR.

---

@ilya-kolchinsky but it is possible you have a DB which is empty?

---

@ashwinb That's correct - it can be that the DB is completely empty, alternatively it can be that all document collections specified by the caller are empty. However, each of these cases would mean the RAG tool is completely non-functional, i.e., it's not capable of answering any query at all. 
Now, providing a misconfigured / non-functioning tool to an agent is clearly something the user should be aware of as it can hardly be the desired behavior. This is also very specific to the RAG tool (most other tools can sometimes return an empty result), thus is cannot be handled at a higher level.
Having written all that, now I realize that the current error message does not capture the needed information - something like "Please check the vector ID / the content of your vector DB" should be added. If we are in agreement here, I will fix the error message.

---

It still seems like this should be a warning instead of an error. If we find that, in practice, most of the times the warning is actually an error, we can elevate it then?

---

@raghotham Yes, that makes sense. But what is the mechanism for returning a warning in LLS API? As of now, we can either raise an error (results in an HTTP error sent back to the user) or return an empty result (again, the user gets an HTTP error). We can log a warning, but the user won't see it.

---

Anyway, this thread is actually out of scope of the PR (which is about a different error). I'd suggest I revert this change so that we don't block the PR and open a separate issue/discussion to converge on this. What do you think?